src/EditorFeatures/Core/Diagnostics/DiagnosticsClassificationTaggerProvider.cs

```diff
@@ -31,7 +31,7 @@ namespace Microsoft.CodeAnalysis.Diagnostics
     [ContentType(ContentTypeNames.RoslynContentType)]
     [ContentType(ContentTypeNames.XamlContentType)]
     [TagType(typeof(ClassificationTag))]
-    internal sealed partial class DiagnosticsClassificationTaggerProvider : AbstractAggregateDiagnosticsTaggerProvider<ClassificationTag>
+    internal sealed partial class DiagnosticsClassificationTaggerProvider : AbstractPushOrPullDiagnosticsTaggerProvider<ClassificationTag>
```

Comment:
this is hte benefit of this PR.  we don't need a PullClassificationDiagTagger and  PushClassificationDiagTagger.  INstead, we just keep all the subclasses the same, and they will either be push or pull without having to do anything special themselves.

Reply:
Would keeping the naming abstract as _AbstractDiagnosticsTaggerProvider_ be reasonable here and across instead of PushOrPull action embedded into it? The direction can come in the naming or as a param as we get to more concrete types.

Reply:
not precisely sure what you mean.  i'm a fan of having it be called AbstractPushOrPullXXX to make it clear that it could be either.  AS to having the subclasses decide, i could... but it would morally be the same (afaict) as hte base type doing the switch (just with more duplication).